backend/de.metas.async/src/main/java/de/metas/async/api/impl/WorkPackageBL.java

```diff
@@ -23,12 +21,8 @@
 			return UserId.optionalOfRepoId(workPackage.getAD_User_InCharge_ID());
 		}
 
-		final I_C_Queue_Block block = workPackage.getC_Queue_Block();
-		Check.assumeNotNull(block, "C_Queue_Block is not null for param 'workPackage'={}", workPackage);
-
-		final I_C_Queue_PackageProcessor packageProcessor = block.getC_Queue_PackageProcessor();
-		Check.assumeNotNull(packageProcessor, "C_Queue_PackageProcessor is not null for block={} of param 'workPackage'={}",
-				block, workPackage);
+		final I_C_Queue_PackageProcessor packageProcessor = workPackage.getC_Queue_PackageProcessor();
```

Comment:
don't use model getters.
u shall get only the package processor ID and then talk with the package processor's Service or Repo.
might be related to https://github.com/metasfresh/metasfresh/pull/12603/files#r823465554